x-pack/plugins/security/public/views/management/roles_grid/components/confirm_delete/confirm_delete.tsx

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiConfirmModal, EuiOverlayMask } from '@elastic/eui';
+import { FormattedMessage, InjectedIntl, injectI18n } from '@kbn/i18n/react';
+import React, { Component, Fragment } from 'react';
+import { toastNotifications } from 'ui/notify';
+import { RolesApi } from '../../../../../lib/roles_api';
+
+interface Props {
+  rolesToDelete: string[];
+  intl: InjectedIntl;
+  callback: (rolesToDelete: string[], errors: string[]) => void;
+  onCancel: () => void;
+}
+
+class ConfirmDeleteUI extends Component<Props, {}> {
+  public deleteUsers = () => {
+    const { rolesToDelete, callback } = this.props;
+    const errors: string[] = [];
+    rolesToDelete.forEach(async roleName => {
+      try {
+        await RolesApi.deleteRole(roleName);
+        toastNotifications.addSuccess(
```

Comment:
nit: should we consider only displaying the success toast once at the end? This could get somewhat "noisy" when deleting a bunch of roles at the same time.

Reply:
I thought that too, but I was trying to preserve the existing functionality.
One counter argument here is that it might be nice to see which operations succeed vs which ones fail. An all-or-nothing toast would potentially hide that information, or get too wordy if we tried to squash those messages together.

Reply:
Fair enough, consider my comment retracted!